collada/__init__.py

```diff
@@ -203,18 +206,44 @@ def __init__(self, filename=None, ignore=None, aux_file_loader=None, zip_filenam
         except ElementTree.ParseError as e:
             raise DaeMalformedError("XML Parsing Error: %s" % e)
 
-        self._loadAssetInfo()
-        self._loadImages()
-        self._loadEffects()
-        self._loadMaterials()
-        self._loadAnimations()
-        self._loadGeometry()
-        self._loadControllers()
-        self._loadLights()
-        self._loadCameras()
-        self._loadNodes()
-        self._loadScenes()
-        self._loadDefaultScene()
+        # if we can't get the current namespace
+        # the tagger from above will use a hardcoded default
+        try:
+            # get the root node, same for both etree and lxml
+            xml_root = self.xmlnode.getroot()
+            if hasattr(xml_root, 'nsmap'):
+                # lxml has an nsmap
+                # use the first value in the namespace map
+                namespace = next(iter(xml_root.nsmap.values()))
+            elif hasattr(xml_root, 'tag'):
+                # for xml.etree we need to extract ns from root tag
+                namespace = xml_root.tag.split('}')[0].lstrip('{')
+            # create a tagging function using the extracted namespace
+            self.tag = tagger(namespace)
+        except BaseException:
+            # failed to extract a namespace, using default
+            traceback.print_exc()
+
+        # functions which will load various things into collada object
+        loaders = [self._loadAssetInfo,
+                   self._loadImages,
+                   self._loadEffects,
+                   self._loadMaterials,
+                   self._loadAnimations,
+                   self._loadGeometry,
+                   self._loadControllers,
+                   self._loadLights,
+                   self._loadCameras,
+                   self._loadNodes,
+                   self._loadScenes,
+                   self._loadDefaultScene]
+        # wrap each loader so if we fail to load something
```

Comment:
This is new behavior, right? I don't like the idea of swallowing exceptions like this, what's the motivation?

Reply:
Yeah, this is new behavior which I'm happy to remove. 

The motivation is that if were are trying to load a scene, and everything works except one light is malformed, this would still return everything that succeeded.

Reply:
I'd rather remove it unless you make it configurable, e.g. like `validate_output` param that's currently there called something like `lenient_parse_mode`. I think it's somewhat surprising behavior that the library would load a malformed file on input by default, especially since that would break round tripping through the library.

Reply:
Sounds good, removed in https://github.com/pycollada/pycollada/pull/86/commits/dd8ac421086d47a2950a901cf3107c54a617d681